src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -857,36 +852,13 @@ markWord ObjectSynchronizer::safe_load_mark(oop object) {
       return dmw;
```

Comment:
Question, why is above guarantee a guarantee, not an assert? You want to be super sure here also in release builds? Do you plan to drop down to assert() eventually?

Reply:
TBH, dunno really. This code originates from ObjectSynchronizer::inflate() and we can probably turn it into an assert.

Reply:
I changed the guarantee to an assert, and will integrate as soon as GHA becomes green, unless you have objections.